packages/hierarchy-builder/src/hierarchy-builder/internal/operators/grouping/GroupHiding.ts

```diff
@@ -0,0 +1,115 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { HierarchyNode } from "../../../HierarchyNode";
+import { GroupingHandlerReturn } from "../Grouping";
+
+/**
+ * @internal
+ */
+export function applyGroupHidingParams(allNodes: HierarchyNode[]): GroupingHandlerReturn {
+  const finalGroupings: GroupingHandlerReturn = { allNodes: [], groupedNodes: [] };
+  for (const node of allNodes) {
+    if (HierarchyNode.isGroupingNode(node)) {
```

Comment:
looks like we could pass it "all nodes" and "grouped nodes" separately to avoid checking if the node is a grouping node?

Reply:
applyGroupHidingParams still needs to return allNodes. So I added a different check, instead I check if the current node in allNodes array is the same as the currentNode in the groupedNodes array, if it is then it might need to be hidden.